DummyServer/ClientApp/src/app/home/home.component.ts

```diff
@@ -0,0 +1,159 @@
+import { Component, Inject } from '@angular/core';
+import { HttpClient, HttpHeaders } from '@angular/common/http';
+import { Subject, Observable } from "rxjs";
+import { switchMap, startWith } from "rxjs/operators";
+
+@Component({
+  selector: 'app-home',
+  templateUrl: './home.component.html',
+})
+export class HomeComponent {
+  public builds: Build[];
+  public branches: Branch[];
+  public definitions: Definition[];
+  public users: User[];
+
+  public selectedDefinition: Definition;
+  public selectedBranch: Branch;
+  public selectedUser: User;
+
+  constructor(private http: HttpClient, @Inject('BASE_URL') private baseUrl: string) {
+
+    this.listenToUpdate<Build[]>(this.refreshBuilds, 'build', r => { this.builds = r; });
+    this.listenToUpdate<Branch[]>(this.refreshBranches,
+      'branch',
+      r => {
+        this.branches = r;
+        this.selectedBranch = this.selectedValueOrDefault(this.selectedBranch, r);
+      });
+    this.listenToUpdate<Definition[]>(this.refreshDefinitions,
+      'definition',
+      r => {
+        this.definitions = r;
+        this.selectedDefinition = this.selectedValueOrDefault(this.selectedDefinition, r);
+      });
+    this.listenToUpdate<User[]>(this.refreshUser,
+      'user',
+      r => {
+        this.users = r;
+        this.selectedUser = this.selectedValueOrDefault(this.selectedUser, r);
+      });
+
+  }
+
+  private listenToUpdate<T>(subject: Subject<Object>, path: string, resultHandler: (result: T) => any) {
+    subject.pipe(
+      startWith(undefined),
+      switchMap(() => this.http.get<T>(this.baseUrl + path))
+    ).subscribe(resultHandler, error => console.error(error));
+  }
+
+  private selectedValueOrDefault<T>(currentValue: T, possibleValues: Array<T>): T {
+    if (!currentValue || possibleValues.indexOf(currentValue) < 0)
+      return this.firstOrDefault(possibleValues);
+  }
+
+  private firstOrDefault<T>(array: Array<T>) {
+    return array.length === 0 ? null : array[0];
+  }
+
+  private refreshBuilds = new Subject();
+  private refreshBranches = new Subject();
+  private refreshDefinitions = new Subject();
+  private refreshUser = new Subject();
+
+  public refresh() {
+    this.refreshBuilds.next();
+  }
+
+  public newBranch(name: string) {
+    var headers = new HttpHeaders().set('Content-Type', 'application/json');
```

Comment:
`var` has some gotchas and should be avoided in typescript: https://www.typescriptlang.org/docs/handbook/variable-declarations.html

Use `let` for variables and `const` where possible

Reply:
ok... why tho